test/boost/chunked_vector_test.cc

```diff
@@ -290,3 +293,147 @@ BOOST_AUTO_TEST_CASE(test_push_back_using_existing_element) {
     do_test_push_back_using_existing_element<chunked_vector_type>([] (chunked_vector_type& v, const push_back_item& x) { v.emplace_back(x); },
             chunked_vector_type::max_chunk_capacity() + 2);
 }
+
+class inject_exceptions {
+    static constexpr bool debug = false;
+
+    ssize_t* objects = nullptr;
+    ssize_t* inject = nullptr;
+
+    void maybe_inject() {
+        if (objects) {
+            if (inject && *objects + 1 >= *inject) {
+                throw std::runtime_error("injected");
+            }
+            ++(*objects);
+        }
+    }
+public:
+    inject_exceptions() = default;
+    explicit inject_exceptions(ssize_t* objects, ssize_t* inject)
+            : objects(objects)
+            , inject(inject)
+    {
+        if (debug) {
+            fmt::print("ctor inject[{}] objects={} inject={}\n", fmt::ptr(this), objects ? *objects : -1, inject ? *inject : -1);
+        }
+        maybe_inject();
+    }
+    inject_exceptions(const inject_exceptions& x)
+            : objects(x.objects)
+            , inject(x.inject)
+    {
+        if (debug) {
+            fmt::print("copy inject[{} <- {}] objects={} inject={}\n", fmt::ptr(this), fmt::ptr(&x), objects ? *objects : -1, inject ? *inject : -1);
+        }
+        maybe_inject();
+    }
+    inject_exceptions(inject_exceptions&& x) noexcept
+            : objects(std::exchange(x.objects, nullptr))
+            , inject(std::exchange(x.inject, nullptr))
+    {
+        if (debug) {
+            fmt::print("move inject[{} <- {}] objects={} inject={}\n", fmt::ptr(this), fmt::ptr(&x), objects ? *objects : -1, inject ? *inject : -1);
+        }
+    }
+    ~inject_exceptions() {
+        if (debug) {
+            fmt::print("dtor inject[{}] objects={} inject={}\n", fmt::ptr(this), objects ? *objects : -1, inject ? *inject : -1);
+        }
+        if (objects) {
+            --(*objects);
+        }
+    }
+};
```

Comment:
I don't understand this class at all. Please document it so readers don't have to reverse-engineer it.